<?xml version="1.0" encoding="utf-8"?>

<launch>
  <arg name="launch_prefix" default=""/>

  <arg name="yolo_weights_path"          default="$(find darknet_ros)/yolo_network_config/weights"/>
  <arg name="yolo_config_path"           default="$(find darknet_ros)/yolo_network_config/cfg"/>

  <arg name="ros_param_file"             default="$(find darknet_ros)/config/{ROS_PARAMS_FNAME}"/>
  <arg name="network_param_file"         default="$(find darknet_ros)/config/{NETWORK_PARAMS_FNAME}"/>

  <rosparam command="load" ns="darknet_ros" file="$(arg ros_param_file)"/>
  <rosparam command="load" ns="darknet_ros" file="$(arg network_param_file)"/>

  <node pkg="darknet_ros" type="darknet_ros" name="{NODE_NAME}" output="screen" launch-prefix="$(arg launch_prefix)">
    <param name="weights_path"          value="$(arg yolo_weights_path)" />
    <param name="config_path"           value="$(arg yolo_config_path)" />
    <param name="yolo_model/weight_file/name" value="{WEIGHTS_FNAME}" />
    <param name="yolo_model/config_file/name" value="{MODEL_FNAME}" />
  </node>
</launch>
